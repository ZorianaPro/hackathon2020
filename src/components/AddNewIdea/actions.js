export const actions = {
	open: 'OPEN:ADDNEWIDEA:OVERLAY',
	close: 'CLOSE:ADDNEWIDEA:OVERLAY',
	submit: 'SUBMIT:ADDNEWIDEA:OVERLAY',
	response: 'RESPONCE:ADDNEWIDEA:OVERLAY',
	error: 'ERROR:ADDNEWIDEA:OVERLAY',
	clean:  'CLEAN:ADDNEWIDEA:OVERLAY',
};

export const open = (content = {}) => ({
	type: actions.open
});

export const close = () => ({
	type: actions.close,
});

export const submit = content => ({
	type: actions.submit,
	value: content
});

export const response = (response) => ({
	type: actions.response,
	value: response
});

export const error = (err) => ({
	type: actions.error,
	value: err
});

export const clean = () => ({
	type: actions.clean
});


export default {
	open,
	close,
	submit,
	response,
	error,
	clean
};
