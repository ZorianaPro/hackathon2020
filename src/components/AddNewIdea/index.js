import React, { useState, useCallback } from 'react';
import Overlay from '../Overlay';
import './AddNewIdea.css';
import '../Overlay/Overlay.css';

const AddNewIdea = ({
  dispatch,
  isOpen,
  errorMessage,
  shouldClose,
  onClose,
  error,
  onSubmit
}) => {
  const [ideaName, setIdeaName] = useState('');
  const [description, setDescription] = useState('');

  const _onClose = useCallback((event) => {
    if (typeof onClose === 'function') {
      onClose(event);
    }
  }, [onClose]);

  const onChange = useCallback((event) => {
    const field = event.target.name;
    const value = event.target.value;
    switch (field) {
      case 'ideaName':
        return setIdeaName(value);
      case 'description':
       return setDescription(value);
    }
  }, []	);

  const _onSubmit = useCallback((event) => {
    event.preventDefault();
      if (typeof onSubmit === 'function') {
        onSubmit(event);
      }
      return false;
  }, []);

  const onFocus = useCallback((event) => {
    const input = event.target;
    const label = input.parentElement.querySelector('label');
    label && label.classList.add('focus');
  }, []);

  const onBlur = useCallback((event) => {
    const input = event.target;
    const label = input.parentElement.querySelector('label');
    if (input.value === '') {
      label && label.classList.remove('focus');
    }
  }, []);

  return (
    <Overlay
      onClose={
        _onClose
      }
      shouldClose={
        shouldClose
      }>
      <div className="AddNewIdea">
        <div className="AddNewIdea-Title">
          Add new idea
        </div>
			  {
				  errorMessage
				  && <div className="Error-Message">
					  { errorMessage }
				  </div>
			  }
			  <form action="" onSubmit={ _onSubmit } method="POST">
				  <div>
					  <label htmlFor="ideaName"
						  className={ `${ideaName !== '' ? 'focus' : ''}` }
					  >
						  Idea Name*
						</label>
					  <input type="text"
					    name="name"
						  onFocus={ onFocus }
						  onBlur={ onBlur }
						  onChange={ onChange }
              required
					  />
					</div>
					<div>
					  <label htmlFor="description"
						  className={ `align-top ${description !== '' ? 'focus' : ''}` }
					  >
						  Idea Description*
						</label>
						<textarea rows="5"
						  name="description"
						  onFocus={ onFocus }
						  onBlur={ onBlur }
						  onChange={ onChange }
              required
						/>
					</div>
					<button type="submit">Submit</button>
				</form>
			</div>
		</Overlay>
	);
};

export default AddNewIdea;