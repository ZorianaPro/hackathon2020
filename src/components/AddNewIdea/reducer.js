import { actions } from "./actions";

const initialState = {
	open: false,
	content: {},
	response: '',
	error: ''
};

export default (state = initialState, action) => {
	switch (action.type) {
		case actions.open:
			return {
				...state,
				open: true
			};
		case actions.close:
			return {
				...state,
				open: false
			};
		case actions.submit:
			return {
				...state,
				content: action.value
			};
		case actions.response:
			return {
				...state,
				response: action.value
			};
		case actions.error:
			return {
				...state,
				error: action.value
			};
		case actions.clean:
			return {
				...state,
				content: {},
				response: '',
				error: ''
			};
		default:
			return state;
	}
};
