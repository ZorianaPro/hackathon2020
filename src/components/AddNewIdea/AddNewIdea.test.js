import React from 'react';
import { shallow, mount } from 'enzyme';
import AddNewIdea from './';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('AddNewIdea', () => {
  const initialState = {
    addNewIdeaReducer: {
      open: true,
      content: {
        id: ''
      }
    }
  };
  const mockStore = configureStore();
  let store;
  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <Provider store={store}><AddNewIdea /></Provider>)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(mount(<Provider store={store}><AddNewIdea /></Provider>)).toMatchSnapshot();
  });
});
