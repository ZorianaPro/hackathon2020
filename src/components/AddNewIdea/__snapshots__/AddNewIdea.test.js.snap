// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewIdea renders as expected 1`] = `
<AddNewIdea>
  <Overlay
    onClose={[Function]}
  >
    <div
      className="Overlay Overlay--isOpening"
    >
      <div
        className="Overlay-Content"
        style={
          Object {
            "animationDuration": "200ms",
          }
        }
      >
        <div
          className="Overlay-Mask"
          onClick={[Function]}
        />
        <div
          className="Overlay-Box"
        >
          <div
            className="Overlay-Close"
            onClick={[Function]}
          >
            <XCloseSVG>
              <svg
                className="XCloseSVG"
              >
                <use
                  xlinkHref="#XCloseSymbol"
                />
              </svg>
            </XCloseSVG>
          </div>
          <div
            className="AddNewIdea"
          >
            <div
              className="AddNewIdea-Title"
            >
              Add new idea
            </div>
            <form
              method="POST"
              onSubmit={[Function]}
            >
              <div>
                <label
                  className=""
                  htmlFor="ideaName"
                >
                  Idea Name*
                </label>
                <input
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                />
              </div>
              <div>
                <label
                  className="align-top "
                  htmlFor="description"
                >
                  Idea Description*
                </label>
                <textarea
                  name="description"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  rows="5"
                />
              </div>
              <button
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </Overlay>
</AddNewIdea>
`;
